visualizer/recommendation.js

```diff
@@ -205,42 +205,30 @@ class Recomendation extends EventEmitter {
       this.articleMenu.append('h2')
         .text('Jump to section')
 
+      const self = this
       this.articleMenu.append('ul')
         .selectAll('li')
         .data(this.readMoreArticle.selectAll('h2').nodes())
         .enter()
         .append('li')
         .text((headerElement) => headerElement.textContent)
+        .attr('id', (headerElement) => headerElement.textContent.replace(/\s/g, ''))
         .on('click', function (headerElement) {
```

Comment:
Just a nit... arrow function here and you can avoid the `self = this` assignment